src/client/reducers/sessionReducer.js

```diff
@@ -17,22 +18,16 @@ const initialState = [
 		playersNeeded: 3, groupId: 6, image: image, sport: "Soccer"},
 	{name: "Water Polo", description: "play time",city: "Vancouver", location: "MacInnes Field", equipment: "Frisbee",
 		playersNeeded: 3, groupId: 7, image: image, sport: "Water Polo"},]
+	}
 
 
-export const createNewSession = (state = initialState, action) => {
+export const sessionReducer = (state = initialState, action) => {
 	switch(action.type) {
         case 'CREATE_NEW_SESSION':
-			state.push(action.payload)
-            return action.payload;
-		default:
-			return state;
-	}
-};
-
-export const fetchSession = (state = initialState, action) => {
-	switch(action.type) {
-		case 'FETCH_SESSIONS':
-			return action.payload;
+			return {
+				...state,
+				sessions: [...state.sessions, action.payload]
```

Comment:
I'm guessing this fixes the error. why is ...state returned seperately in line 28, didn't spot an explicit use for it.

Reply:
It doesn't have a use right now. I left this here in case we want to add more properties to this state in the future. Currently it does nothing.